x-pack/plugins/ml/server/saved_objects/initialization/initialization.ts

```diff
@@ -35,21 +41,25 @@ export function jobSavedObjectsInitializationFactory(core: CoreStart, spacesEnab
         return;
       }
 
+      if ((await _needsInitializing(savedObjectsClient)) === false) {
+        // ml job saved objects have already been initialized
+        return;
+      }
+
       const jobSavedObjectService = jobSavedObjectServiceFactory(
         savedObjectsClient,
         savedObjectsClient,
         spacesEnabled,
+        security?.authz,
         () => Promise.resolve() // pretend isMlReady, to allow us to initialize the saved objects
       );
 
-      if ((await _needsInitializing(savedObjectsClient)) === false) {
-        // ml job saved objects have already been initialized
-        return;
-      }
-
       mlLog.info('Initializing job saved objects');
+      // create space overrides for specific jobs
+      const jobSpaceOverrides = await createJobSpaceOverrides(client);
+      // initialize jobs
       const { initSavedObjects } = repairFactory(client, jobSavedObjectService);
-      const { jobs } = await initSavedObjects();
+      const { jobs } = await initSavedObjects(false, jobSpaceOverrides);
       mlLog.info(`${jobs.length} job saved objects initialized for * space`);
```

Comment:
Nit: now that the space can be overridden per job this message might be misleading.

Reply:
Updated in https://github.com/elastic/kibana/pull/83127/commits/dd59803f23f24990b038ae5196ec5a9896190566